test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/MutableRealmTests.kt

```diff
@@ -92,14 +92,34 @@ class MutableRealmTests {
     }
 
     @Test
-    fun set_throwsOnDuplicatePrimaryKey() {
+    fun set_updatesExistingObjectInTree() {
+        val child = SampleWithPrimaryKey().apply {
+            primaryKey = 1
+            stringField = "INIT"
+        }
+        val parent = realm.writeBlocking {
+            copyToRealm(
+                SampleWithPrimaryKey().apply {
+                    primaryKey = 2
+                    nullableObject = child
+                }
+            )
+        }
+        realm.query<SampleWithPrimaryKey>("primaryKey = '1'").find().single().run {
+            assertEquals("INIT", stringField)
+        }
+
+        child.apply {
+            stringField = "UPDATE"
+        }
         realm.writeBlocking {
-            val sample = copyToRealm(SampleWithPrimaryKey())
-            assertFailsWith<IllegalArgumentException> {
-                sample.nullableObject = SampleWithPrimaryKey()
+            findLatest(parent)!!.apply {
+                nullableObject = child
```

Comment:
The code changes seems to have a few more code paths, e.g. realmObject, embeddedobject, embeddedWithKey... We should probably also test these. But since we didn't do that in original code, we can probably just do that in another PR.

Reply:
There are test for the three types of assignments that we can do:
- RealmObject is this test
- EmbeddedRealmObject is in https://github.com/realm/realm-kotlin/pull/864/files#diff-e211bb2185cacfa97f056a11aaf87c881363ea9add72952fe2305adaafe6afc8R244
- RealmList<RealmObject> is in https://github.com/realm/realm-kotlin/pull/864/files#diff-bb43ea504213fcf23891f060506ce3837d2812808acb02286a49185f54883755R335

I guess it isn't tested for `RealmList<EmbeddedRealmObject>` but the codepath for that is not different from `RealmList<RealmObject>` so the above test should already verify that we get the UpdatePolicy propagated correctly.